tests/js/preferencesRouter-tests.js

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/fluid-project/first-discovery-server/raw/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = fluid || require("infusion");
+var gpii = fluid.registerNamespace("gpii");
+var jqUnit       = require("jqUnit");
+fluid.registerNamespace("gpii.tests.firstDiscovery.server");
+
+require("../../src/js/firstDiscoveryServer.js");
+require("../../node_modules/gpii-express/tests/js/lib/test-helpers.js");
+require("../../node_modules/kettle");
+require("../../node_modules/kettle/lib/test/KettleTestUtils");
```

Comment:
These direct "blind includes" into other packages are all unstable and shouldn't be used - they will break in future versions of npm where module layout can't be predicted.

The latter two lines should be replaced by
var kettle = fluid.registerNamespace("kettle");
kettle.loadTestingSupport();

the first require is fine since it is a file in the current module.
the require of test-helpers probably can't be replaced now but I've just filed https://issues.gpii.net/browse/GPII-1302 which you can reference here and fix up the load once it is resolved